### ISSUE TITLE
Nice error for old git versions. Documentation of version requirements.

### ISSUE THREAD

**Issue by tbenthompson** [pypa/setuptools-scm#540]
I just upgraded a CI server from setuptools_scm==5.0.2 to setuptools_scm==6.0.0 and immediately ran into errors due to:
```
git log -n 1 HEAD --format=%cI
```
not being valid on our git version 1.8.3.1. Since that git version is almost six years old now, I don't expect support, but I wanted to suggest adding some documentation to the README and perhaps a git version check into the code. 

Currently the above command leads to the error:
```
        raise ValueError("time data %r does not match format %r" %
    ValueError: time data '%cI' does not match format '%Y-%m-%d'
```
It would be much nicer to see an error saying that my git version is not supported. 

Thanks very much!

**Comment by RonnyPfannschmidt** [comment]
this was fixed on version 6.0.1 see #537 and #538

**Comment by tbenthompson** [comment]
Ah thank you!
